isis/src/base/objs/Cube/Cube.cpp

```diff
@@ -2133,6 +2134,96 @@ namespace Isis {
   }
 
 
+/**
+ * Returns the latitude and longitude range for the Cube. More accurate than the minimum and 
+ * maximum latitude and longitude from the mapping group. 
+ * 
+ * @param minLatitude minimum latitude of the cube
+ * @param maxLatitude maximum latitude present in the cube
+ * @param minLongitude minimum longitude present in the cube
+ * @param maxLongitude maximum longitude present in the cube
+ */
+  void Cube::latLonRange(double &minLatitude, double &maxLatitude, double &minLongitude, double &
+                         maxLongitude) {
+    Camera *cam;
+    TProjection *proj;
+
+    bool isGood = false;
+    bool noCamera;
+
+    if (hasGroup("Instrument")) {
+      noCamera = false;
+    }
+    else {
+      noCamera = true;
+    }
+
+    // setup camera or projection
+    if (noCamera) {
+     try {
+       proj = (TProjection *) projection();
+     }
+     catch(IException &e) {
+       QString msg = "Cannot calculate lat/lon range without a camera or projection";
+       throw IException(e, IException::User, msg, _FILEINFO_);
+     }
+    }
+    else {
+      try {
+        cam = camera();
+      }
+      catch(IException &e) {
+        QString msg = "Unable to create camera when calculating a lat/lon range.";
+        throw IException(e, IException::User, msg, _FILEINFO_);
+      }
+    }
+
+    // Iterate over all samp/line combos in cube
+    minLatitude = 99999;
+    minLongitude = 99999;
+    maxLatitude = -99999;
+    maxLongitude = -99999;
+
+    for (int sample = 0; sample < sampleCount(); sample++) {
+    // Checks to see if the point is in outer space
+      for (int line = 0; line < lineCount(); line++) {
```

Comment:
This is potentially a very time consuming process. Something like a quad tree approach would probably work significantly faster but may be a little less accurate.

Reply:
I don't think a quad tree helps you here, because you would have to get bounds to do the construction. A sampling approach might though - take the min/max pixels and then randomly sample n additional pixels to get an approximate lat/lon range. In most cases this will be the correct values. To account for weird geometries, you could sample without replacement in chunks of n, if the min/max values do not change after chunk m, break.